test/modules/linear_boltzmann_solvers/transport_steady/reed_balance.lua

```diff
@@ -0,0 +1,100 @@
+-- Standard Reed 1D 1-group problem
+
+-- Create Mesh
+widths = {2., 1., 2., 1., 2.}
+nrefs  = {200,200,200,200,200}
+
+Nmat = #widths
+
+nodes = {}
+counter = 1
+nodes[counter] = 0.
+for imat = 1, Nmat do
+    dx = widths[imat]/nrefs[imat]
+    for i = 1, nrefs[imat] do
+        counter = counter + 1
+        nodes[counter] = nodes[counter-1] + dx
+    end
+end
+
+meshgen = mesh.OrthogonalMeshGenerator.Create({ node_sets = {nodes} })
+mesh.MeshGenerator.Execute(meshgen)
+
+-- Set Material IDs
+z_min = 0.0
+z_max = widths[1]
+for imat = 1, Nmat do
+    z_max = z_min + widths[imat]
+    print('imat, zmin, zmax', imat,z_min,z_max)
+    lv = logvol.RPPLogicalVolume.Create(
+                {
+                    infx=true,
+                    infy=true,
+                    zmin=z_min, zmax=z_max
+                }
+    )
+    mesh.SetMaterialIDFromLogicalVolume(lv, imat-1)
+    z_min = z_max
+end
+
+-- Create materials
+mat_names = { "AbsoSrc", "Abso", "Void", "ScatSrc", "Scat"}
+materials = {}
+for imat = 1, Nmat do
+    materials[imat] = mat.AddMaterial(mat_names[imat])
+end
+
+-- Add cross sections to materials
+total = { 50., 5., 0., 1. , 1.  }
+c     = {  0., 0., 0., 0.9, 0.9 }
+for imat = 1, Nmat do
+    mat.AddProperty(materials[imat], TRANSPORT_XSECTIONS)
+    mat.SetProperty(materials[imat], TRANSPORT_XSECTIONS, SIMPLE_ONE_GROUP, total[imat], c[imat])
```

Comment:
Not for this PR, but what does anyone like the `AddProperty` then `SetProperty`? IMO, `AddProperty` and `SetProperty` should have the same call signatures and differ only by whether they actually create a property on a material or not. For example, `AddProperty` should throw an error if the property already exists on the material and `SetProperty` should throw an error if the property does not exist on the material. If there is agreement here, I can create an issue.

Reply:
I do not like the Add followed by a Set later on. I think we discussed this some time ago. I must be reading this incorrectly, but I didn't fully follow the rest of your message. I would get rid of AddProperty altogether and use SetProperty only. If the property already exists, then throw error.

Reply:
Currently, `AddProperty` just creates an empty material property and `SetProperty` overwrites an existing property. In that paradigm, you should only call `AddProperty` if the property type does not exist on a material and `SetProperty` if the property type already exists. There are several applications where one may want to swap material properties (step insertion, brute force search for inverse problems, etc.), so I don't think `SetProperty` should throw an error if the property already exists. See #252.